frontend/src/utils/convertSignatureOutput.ts

```diff
@@ -0,0 +1,82 @@
+import getStroke from 'perfect-freehand'
+
+import {
+  boxHeightDefault,
+  boxWidthDefault,
+  getBoundingBox,
+  signatureOutputPaddingDefault,
+  signatureStrokeSize,
+  signatureStrokeSmoothing,
+  signatureStrokeStreamline,
+  signatureStrokeThinning,
+} from '~shared/utils/signature'
+
+export const drawStroke = (
+  ctx: CanvasRenderingContext2D,
+  points: number[][],
+): void => {
+  ctx.beginPath()
+  ctx.moveTo(points[0][0], points[0][1])
+
+  for (let i = 1; i < points.length; i++) {
+    const [x, y] = points[i]
+    ctx.lineTo(x, y)
+  }
+
+  ctx.closePath()
+  ctx.fill()
+}
+
+export const convertToSignatureSvgString = (
+  vectorArray: [number, number, number][][],
+  padding = signatureOutputPaddingDefault,
+): string => {
+  if (vectorArray.length === 0) return ''
+
+  const { minX, minY, maxX, maxY } = getBoundingBox(vectorArray)
+  const boxWidth = maxX - minX || boxWidthDefault
+  const boxHeight = maxY - minY || boxHeightDefault
+
+  const canvasWidth = boxWidth + 2 * padding
+  const canvasHeight = boxHeight + 2 * padding
+
+  const paths = vectorArray.map((stroke) => {
+    if (stroke.length === 0) return ''
+
+    const normalizedStroke = stroke.map(([x, y, pressure]) => [
+      x - minX + padding,
+      y - minY + padding,
+      pressure,
+    ])
+
+    const strokePoints = getStroke(normalizedStroke, {
+      size: signatureStrokeSize,
+      thinning: signatureStrokeThinning,
+      smoothing: signatureStrokeSmoothing,
+      streamline: signatureStrokeStreamline,
+    })
+
+    if (strokePoints.length === 0) return ''
+
+    const pathData = strokePoints
+      .map(([x, y], i) =>
+        i === 0
+          ? `M ${x.toFixed(2)} ${y.toFixed(2)}`
+          : `L ${x.toFixed(2)} ${y.toFixed(2)}`,
+      )
+      .join(' ')
+
+    return `<path d="${pathData} Z" fill="black" />`
+  })
+
+  const svg = `
+    <svg xmlns="http://www.w3.org/2000/svg"
+         width="${canvasWidth}" height="${canvasHeight}"
+         viewBox="0 0 ${canvasWidth} ${canvasHeight}">
+      <rect width="100%" height="100%" fill="white" />
+      ${paths.join('\n')}
+    </svg>
+  `.trim()
```

Comment:
Hm, briefly looked through for XSS attacks on `<svg />` seems like we have control of the input values (for now). Wondering if the inputs could be strongly validated or sanitised?

Wouldn't want such things to be injected where `paths = ["<script>alert(1)</script>", ...]`.


Perhaps something along the lines of this?
```

import sanitize from 'sanitize-html'

...

const cleanPaths = sanitizeHtml(paths.join('\n'))

...

    <svg xmlns="http://www.w3.org/2000/svg"
         width="${canvasWidth}" height="${canvasHeight}"
         viewBox="0 0 ${canvasWidth} ${canvasHeight}">
      <rect width="100%" height="100%" fill="white" />
      ${cleanPaths}
    </svg>

```